python/src/table.rs

```diff
@@ -303,4 +303,17 @@ impl Table {
             })
         })
     }
+
+    pub fn migrate_manifest_paths_v2(self_: PyRef<'_, Self>) -> PyResult<Bound<'_, PyAny>> {
+        let inner = self_.inner_ref()?.clone();
+        future_into_py(self_.py(), async move {
+            inner
+                .as_native()
+                .ok_or_else(|| PyValueError::new_err("This cannot be run on a remote table"))?
+                .migrate_manifest_paths_v2()
+                .await
+                .infer_error()
+                .map(|_| ())
```

Comment:
What's getting mapped away here?  Doesn't the function already return `Result<()>`?